### PR TITLE
Makes explosions not take forever, fixes some pipe deletion stuff

### PR DESCRIPTION
**Pull request by Tastyfish**
Makes explosions faster by preventing the pipenets and powernets from rebuilding 1000 times on an explosion.
 - This is done by deferring the rebuilds to happen in the pipenet or machine process, respectively.
 - I also fixed a few places where atmos stuff sometimes runtimes due to rapid deletion of stuff. (Some of the runtimes happened even on master.)
  - This could incidentally fix some bugs but also cause new ones, who knows.

## Some maxcap explosion times
`113,135,1` is the bridge.
`131,77,1` is atmosia (where things really go on fire :wink:).
`143,108,1` is medbay.
### master
Explosion(113,135,1)(d5,h10,l20): Took 2.2 seconds.
Explosion(131,77,1)(d5,h10,l20): Took 3.5 seconds.
Explosion(143,108,1)(d5,h10,l20): Took 1.1 seconds.

Explosion(113,135,1)(d5,h10,l20): Took 1.8 seconds.
Explosion(131,77,1)(d5,h10,l20): Took 3.7 seconds.
Explosion(143,108,1)(d5,h10,l20): Took 1.1 seconds.

### this pr
Explosion(113,135,1)(d5,h10,l20): Took 0.6 seconds.
Explosion(131,77,1)(d5,h10,l20): Took 0.5 seconds.
Explosion(143,108,1)(d5,h10,l20): Took 0.5 seconds.

Explosion(113,135,1)(d5,h10,l20): Took 0.4 seconds.
Explosion(131,77,1)(d5,h10,l20): Took 0.5 seconds.
Explosion(143,108,1)(d5,h10,l20): Took 0.5 seconds.

### CODE REVIEW

**Comment by QQCucumber** [comment]
Oh my god, this is amazing. But I think what also causes explosion lag is the gibs colliding with people - escape was once brought to a standstill for several minutes, as it processed me being hit in the legs and chest with kindeys and livers until finally I gibbed too. Would it be possible to streamline that as well?

**Comment by Tastyfish** [comment]
Hm, not sure, since that's physical things colliding in real time.

**Comment by QQCucumber** [comment]
Would it be possible to have it 'skip' people who are about to be gibbed by the explosion anyway? Like, what's the point in registering the damage of livers and kidneys hitting them if they're a millisecond away (of ingame time) away from being gibbed as well?

File: code/ATMOSPHERICS/pipes/manifold.dm [hunk 1]
```diff
@@ -74,23 +74,20 @@
...
+		node2.defer_build_network()
 		node2 = null
-		A.build_network()
 	if(node3)
-		var/obj/machinery/atmospherics/A = node3
 		node3.disconnect(src)
+		node3.defer_build_network()
 		node3 = null
-		A.build_network()
-	releaseAirToTurf()
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/pipes/manifold.dm]: Why get rid of all the air releasing to turfs?

Comment (Tastyfish) [inline code/ATMOSPHERICS/pipes/manifold.dm]: They happen in pipe/Destroy() already; it was getting called twice.

Comment (Fox-McCloud) [inline code/ATMOSPHERICS/pipes/manifold.dm]: Ahh, gotcha. Thanks.

Comment (Tastyfish) [inline code/ATMOSPHERICS/pipes/manifold.dm]: :wink:

**Comment by Bigtimetripod** [comment]
Bombed escape soo hard few days ago with 2 maxcaps that server hung for a good 4-7 minutes. I was afraid that shuttle is gonna leave before the gameplay gets smooth again.

Then was a problem that escape has 20-30 people... all them gibs and crap like take quite a while.

**Pull request merged**